src/components/Tool/Appeal/Modals/AddContactToAppealModal/AddContactToAppealModal.tsx

```diff
@@ -39,33 +46,47 @@ export const AddContactToAppealModal: React.FC<
     AppealsContext,
   ) as AppealsType;
 
-  const { data } = useAppealQuery({
+  const { data } = useAppealContactsQuery({
     variables: {
       accountListId: accountListId ?? '',
       appealId: appealId ?? '',
     },
   });
 
   const existingContactIds = data?.appeal?.contactIds ?? [];
+  const excludedContacts = data?.appeal?.excludedAppealContacts ?? [];
+
+  const onSubmit = async ({ contactIds }: AddContactFormikSchema) => {
+    const excludedContactIds = excludedContacts.map(
+      (excludedContact) => excludedContact.contact?.id,
+    );
+    const addingExcludedContact = contactIds.some((contactId) =>
+      excludedContactIds.includes(contactId),
+    );
+
+    const attributes: Attributes = {
+      id: appealId,
+      contactIds: [...existingContactIds, ...contactIds],
+    };
+
+    if (addingExcludedContact) {
+      attributes.forceListDeletion = true;
+    }
```

Comment:
What do you think about this?

```suggestion
    const attributes: Attributes = {
      id: appealId,
      contactIds: [...existingContactIds, ...contactIds],
      forceListDeletion: addingExcludedContact,
    };
```